cvat-ui/src/components/labels-editor/label-form.tsx

```diff
@@ -502,8 +502,11 @@ export default class LabelForm extends React.Component<Props> {
                     },
                     {
                         validator: (_rule: any, labelName: string) => {
+                            if (!labelName.trim()) {
+                                return Promise.reject(new Error('Label name cannot be empty'));
+                            }
```

Comment:
Im not sure your fix for resetting the constructor works well in case if we edit already existing label. See the screenshot attached - we got two messages. There is already a required rule introduced in this form item, in lines above, its better to use it properly instead of trying to mimic required in validator. In our case I belive we should enable even if label is not in props.
![image](https://github.com/opencv/cvat/assets/50956430/ebaaf8d1-5cb6-4698-9b51-76ce6d145202)